kroxylicious-kms-provider-aws-kms/src/main/java/io/kroxylicious/kms/provider/aws/kms/AwsKmsService.java

```diff
@@ -20,14 +21,28 @@
 @Plugin(configType = Config.class)
 public class AwsKmsService implements KmsService<Config, String, AwsKmsEdek> {
 
-    @NonNull
+    @SuppressWarnings("java:S3077") // Config is an immutable object
+    private volatile Config config;
+
     @Override
-    public AwsKms buildKms(Config options) {
-        return new AwsKms(options.endpointUrl(),
-                options.accessKey().getProvidedPassword(),
-                options.secretKey().getProvidedPassword(),
-                options.region(),
-                Duration.ofSeconds(20), options.sslContext());
+    public void initialize(@NonNull Config config) {
+        Objects.requireNonNull(config);
+        if (this.config != null) {
+            throw new IllegalStateException("KMS service is already initialized");
+        }
```

Comment:
Why enforce this? 

I agree we don't really want to do it multiple times but I don't see why the AwsKmsService enforces this.